fuzzers/libafl-fuzz/src/afl_stats.rs

```diff
@@ -0,0 +1,603 @@
+        self.last_find = current_time();
+    }
+
+    fn maybe_update_last_crash(
+        &mut self,
+        testcase: &Testcase<<<E as UsesState>::State as UsesInput>::Input>,
+        state: &E::State,
+    ) {
+        if testcase
+            .hit_objectives()
+            .contains(&Cow::Borrowed("CrashFeedback"))
+        {
+            self.last_crash = current_time();
+            self.execs_at_last_objective = *state.executions();
+        }
+    }
+
+    fn maybe_update_last_hang(
+        &mut self,
+        testcase: &Testcase<<<E as UsesState>::State as UsesInput>::Input>,
+        state: &E::State,
+    ) {
+        if testcase
+            .hit_objectives()
+            .contains(&Cow::Borrowed("TimeoutFeedback"))
+        {
+            self.last_hang = current_time();
+            self.execs_at_last_objective = *state.executions();
+        }
+    }
+
+    fn check_interval(&mut self) -> bool {
+        let cur = current_time();
+        if cur.checked_sub(self.last_report_time).unwrap_or_default() > self.stats_report_interval {
+            self.last_report_time = cur;
+            return true;
+        }
+        false
+    }
+    fn maybe_update_cycles(&mut self, queue_cycles: u64) {
+        if queue_cycles > self.cycles_done {
+            self.cycles_done += 1;
+        }
+    }
+
+    fn maybe_update_cycles_wo_finds(&mut self, queue_cycles: u64) {
+        if queue_cycles > self.cycles_done && self.last_find < current_time() {
+            self.cycles_wo_finds += 1;
+        }
+    }
+
+    #[allow(clippy::cast_precision_loss)]
+    #[allow(clippy::unused_self)]
+    fn calculate_stability(&self, unstable_entries: usize, filled_entries: u64) -> f64 {
+        ((filled_entries as f64 - unstable_entries as f64) / filled_entries as f64) * 100.0
+    }
+}
+
+impl Display for AFLPlotData<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{},", self.relative_time)?;
+        write!(f, "{},", self.cycles_done)?;
+        write!(f, "{},", self.cur_item)?;
+        write!(f, "{},", self.corpus_count)?;
+        write!(f, "{},", self.pending_total)?;
+        write!(f, "{},", self.pending_favs)?;
+        write!(f, "{},", self.total_edges)?;
+        write!(f, "{},", self.saved_crashes)?;
+        write!(f, "{},", self.saved_hangs)?;
+        write!(f, "{},", self.max_depth)?;
+        write!(f, "{},", self.execs_per_sec)?;
+        write!(f, "{},", self.execs_done)?;
+        write!(f, "{}", self.edges_found)?;
+        Ok(())
+    }
+}
+impl AFLPlotData<'_> {
+    fn get_header() -> String {
+        "# relative_time, cycles_done, cur_item, corpus_count, pending_total, pending_favs, total_edges, saved_crashes, saved_hangs, max_depth, execs_per_sec, execs_done, edges_found".to_string()
+    }
+}
+impl Display for AFLFuzzerStats<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "start_time        : {}", &self.start_time)?;
```

Comment:
I wonder if it's faster to format this and then do one big write, since every writeln may trigger a syscall

Reply:
As in, we coudl make this one a big
```rust
writeln!(f, "...
...
...
", ...)
```

a bit uglier to read but potentially faster(?)

Reply:
Will investigate